### ISSUE TITLE
Error thrown when using `numpy` version 2.*

### ISSUE THREAD

**Issue by hpretl** [dreoilin/pygmid#28]
When using `numpy` version 2.*, the following error is thrown:

```
File /usr/local/lib/python3.12/site-packages/pygmid/Lookup.py:261, in Lookup._RatioVRatioLK(self, outkeys, varkeys, vararg, pars, **ipkwargs)
    [257](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/pygmid/Lookup.py:257) dim = x.shape
    [258](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/pygmid/Lookup.py:258) output = np.zeros((dim[1], len(xdesired)))
    [259](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/pygmid/Lookup.py:259) ipkwargs = {
    [260](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/pygmid/Lookup.py:260)     'kind' : pars['METHOD'],
...
    [404](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/numpy/__init__.py:404)         "`np.chararray` is deprecated and will be removed from "
    [405](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/numpy/__init__.py:405)         "the main namespace in the future. Use an array with a string "
    [406](https://file+.vscode-resource.vscode-cdn.net/usr/local/lib/python3.12/site-packages/numpy/__init__.py:406)         "or bytes dtype instead.", DeprecationWarning, stacklevel=2)

AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.
```

Please update `pygmid` to also support latest `numpy`.

**Comment by dreoilin** [comment]
Updated to remove references to NaN and fix compatibility issues with numpy 2.0 .

`pip install --upgrade pygmid`

to install v1.2.12.

Cian

**Comment by hpretl** [comment]
Excellent, thank you!
